src/client/BaseClient.ts

```diff
@@ -5,6 +5,12 @@ import { TimerManager } from '@klasa/timer-manager';
 
 import { BaseClientOptionsDefaults } from '../util/Constants';
 
+const plugins: Set<Function> = new Set();
```

Comment:
Alternatively, you can do this:
```suggestion
const plugins = new Set<Function>();
```

Reply:
plugins should be a true private static property of the client. I am also unsure if this pr belongs on BaseClient or Client. Others chime in with arguments for either.

Reply:
Actually, I am pretty sure it has to be on Client because of Piece Store access.